tests/models/test_transformers_vision.py

```diff
@@ -0,0 +1,235 @@
+
+
+def test_transformers_vision_wrong_number_images(model, images):
+    with pytest.raises(ValueError):
+        a = model(
+            {
+                "prompts": "<image>Describe this image in one sentence:",
+                "images": [images[0], images[1]],
+            }
+        )
+        print(a)
+
+
+def test_transformers_vision_wrong_input_type(model):
+    with pytest.raises(NotImplementedError):
+        model.generate("invalid input", None)
+
+
+def test_transformers_inference_kwargs(model, images):
+    result = model(
+        {"prompts": "<image>Describe this image in one sentence:", "images": images[0]},
+        max_new_tokens=100,
+    )
+    assert isinstance(result, str)
+
+
+def test_transformers_invalid_inference_kwargs(model, images):
+    with pytest.raises(ValueError):
+        model(
+            {
+                "prompts": "<image>Describe this image in one sentence:",
+                "images": images[0],
+            },
+            foo="bar",
+        )
+
+
+def test_transformers_several_images(model, images):
+    result = model(
+        {
+            "prompts": "<image><image>Describe this image in one sentence:",
+            "images": [images[0], images[1]],
+        }
+    )
+    assert isinstance(result, str)
+
+
+def test_transformers_vision_json(model, images):
+    class Foo(BaseModel):
+        name: str
+
+    result = model(
+        {"prompts": "<image>Give a name to this animal.", "images": images[0]},
+        Json(Foo),
+    )
+    assert "name" in result
+
+
+def test_transformers_vision_regex(model, images):
+    result = model(
+        {"prompts": "<image>How old is it?", "images": images[0]}, Regex(r"[0-9]")
+    )
+
+    assert isinstance(result, str)
+    assert re.match(r"[0-9]", result)
+
+
+def test_transformers_vision_choice(model, images):
+    class Foo(Enum):
+        cat = "cat"
+        dog = "dog"
+
+    result = model(
+        {"prompts": "<image>Is it a cat or a dog?", "images": images[0]}, Choice(Foo)
+    )
+
+    assert isinstance(result, str)
+    assert result in ["cat", "dog"]
+
+
+def test_transformers_vision_batch_samples(model, images):
+    result = model(
+        {"prompts": "<image>Describe this image in one sentence.", "images": images[0]},
+        num_return_sequences=2,
+        num_beams=2,
+    )
+    assert isinstance(result, list)
+    assert len(result) == 2
+    result = model(
+        {
+            "prompts": [
+                "<image>Describe this image in one sentence.",
+                "<image>Describe this image in one sentence.",
+            ],
+            "images": [images[0], images[0]],
+        }
+    )
+    assert isinstance(result, list)
+    assert len(result) == 2
+    result = model(
+        {
+            "prompts": [
+                "<image>Describe this image in one sentence.",
+                "<image>Describe this image in one sentence.",
+            ],
+            "images": [images[0], images[0]],
```

Comment:
Q: Is this batching?

Reply:
Yes. This part tests batching (2) + multiple samples (2). There are 2 images, one for each batch. We check that the output is a list of 2 lists each containing 2 elements.